src/core/mac/mac.cpp

```diff
@@ -980,111 +980,61 @@ extern "C" void otPlatRadioTxDone(otInstance *aInstance, otRadioFrame *aFrame, o
     else
 #endif // OPENTHREAD_ENABLE_RAW_LINK_API
     {
-        instance->GetThreadNetif().GetMac().TransmitDoneTask(aFrame, aAckFrame, aError);
+        instance->GetThreadNetif().GetMac().HandleTransmitDone(aFrame, aAckFrame, aError);
     }
 
 exit:
     otLogFuncExit();
 }
 
-void Mac::TransmitDoneTask(otRadioFrame *aFrame, otRadioFrame *aAckFrame, otError aError)
+void Mac::HandleTransmitDone(otRadioFrame *aFrame, otRadioFrame *aAckFrame, otError aError)
 {
-    Frame *txFrame = static_cast<Frame *>(aFrame);
-    Address addr;
+    Frame &sendFrame = *static_cast<Frame *>(aFrame);
+    Address dstAddr;
     bool framePending = false;
+    Neighbor *neighbor = NULL;
 
+    // Stop the ack timer.
     mMacTimer.Stop();
 
-    txFrame->GetDstAddr(addr);
-
-    if (aError == OT_ERROR_NONE && txFrame->GetAckRequest() && aAckFrame != NULL)
```

Comment:
Adding some comments to help with the review (since diff looks bigger than what's actually being changed in this PR). 

The logic to get the "frame-pending" bit and update the neighbor RSS is moved after the error handling (checking for CSMA retry and a full frame re-tx attempt).